src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -695,19 +710,24 @@ function VisualizeAppController(
     );
   }
 
-  const unlinkFromSavedSearch = () => {
-    const searchSourceParent = savedSearch.searchSource;
+  $scope.unlinkFromSavedSearch = () => {
+    const searchSourceParent = searchSource.getParent();
     const searchSourceGrandparent = searchSourceParent.getParent();
     const currentIndex = searchSourceParent.getField('index');
 
     searchSource.setField('index', currentIndex);
     searchSource.setParent(searchSourceGrandparent);
 
+    delete savedVis.savedSearchId;
```

Comment:
I would suggest you to keep the `editor.js` file as clean as possible to avoid corruptions (a lot of stuff is not covered with tests).
Deletion of these fields is handled in `handleLinkedSearch`.